test/test-fetch.sh

```diff
@@ -80,6 +80,13 @@ begin_test "fetch"
   assert_local_object "$contents_oid" 1
   assert_local_object "$b_oid" 1
 
+  # test with commit sha1 specified
+  rm -rf .git/lfs/objects
+  newbranch_sha1=$(git rev-parse newbranch)
+  git lfs fetch origin "$newbranch_sha1"
+  assert_local_object "$contents_oid" 1
+  assert_local_object "$b_oid" 1
```

Comment:
done in https://github.com/github/git-lfs/pull/1323/commits/7c246de3d6b268eb8ff6c87f47a07d126ddf05ab (if I correctly understood your comment)